crates/parser/src/grammar/paths.rs

```diff
@@ -54,7 +54,7 @@ fn path_for_qualifier(
     mut qual: CompletedMarker,
 ) -> CompletedMarker {
     loop {
-        let use_tree = matches!(p.nth(2), T![*] | T!['{']);
+        let use_tree = mode == Mode::Use && matches!(p.nth(2), T![*] | T!['{']);
```

Comment:
What is the purpose of this change?

Reply:
This improves the ast in case of a syntax error and recovers better imo.

| Before | After |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/5961244/181785329-918b38bc-4bd5-44a2-95fa-4888ee70bfad.png) | ![image](https://user-images.githubusercontent.com/5961244/181785874-388b12ca-67ba-40b5-99a0-20aaaf129ac9.png) |

I initially thought that this was required so completions can work but later on I found out that completions actually "injects `intellijRulezz` into ast". So this better parsing recovery isn't actually needed but I thought it would be an overall improvement. But I can also leave it out, I don't mind either way.

Reply:
Aight, this change seems fine to me, and I do think that's actually needed to work for this here looking at the parse trees